src/cli_driver.rs

```diff
@@ -449,14 +448,30 @@ impl ProcessingSession {
             };
 
             if let Some(par) = tex_path.parent() {
-                fs_root = par;
+                output_path = par;
                 io_builder.filesystem_root(par);
             } else {
                 return Err(ErrorKind::Msg(format!("can't figure out a parent directory for input path \"{}\"",
                                                   tex_path.to_string_lossy())).into());
             }
         }
 
+        if let Some(dir) = args.value_of("outdir") {
```

Comment:
This should use [value_of_os](https://docs.rs/clap/2.25.0/clap/struct.ArgMatches.html#method.value_of_os) to get an OsString from the argument.